.github/workflows/bencher.yml

```diff
@@ -31,26 +55,53 @@ jobs:
           fetch-depth: 0
       - uses: actions/download-artifact@v4
         with:
-          name: release-binary-linux
-          path: release-binary-linux
-      - name: unPackage binary
-        run: tar -xzf release-binary-linux/target.tar.gz
+          name: ${{ inputs.profile }}-binary-${{ inputs.target }}
+          path: ${{ inputs.profile }}-binary-${{ inputs.target }}
+      # Linux and macOS uploads compressed binary
+      - name: unPackage binary (Linux or macOS)
+        if: ${{ inputs.target == 'linux' || inputs.target == 'macos' }}
+        run: tar -xzf ${{ inputs.profile }}-binary-${{ inputs.target }}/target.tar.gz
+      - name: File location (Linux or macOS)
+        if: ${{ inputs.target == 'linux' || inputs.target == 'macos' }}
+        run: echo "SERVO_BINARY_PATH=target/${{ inputs.profile }}/servo" >> $GITHUB_ENV
+      - name: File location (Windows)
+        if: ${{ inputs.target == 'windows' }}
+        run: echo "SERVO_BINARY_PATH=${{ inputs.profile }}-binary-windows/Servo.exe" >> $GITHUB_ENV
+      - name: File location (Android)
+        if: ${{ contains(inputs.target, 'android') }}
+        run: echo "SERVO_BINARY_PATH=${{ inputs.profile }}-binary-android/servoapp.apk" >> $GITHUB_ENV
+      - name: File location (Open Harmony, signed)
+        continue-on-error: true
+        if: ${{ contains(inputs.target, 'ohos') && env.SERVO_OHOS_SIGNING_CONFIG != '' }}
+        run: echo "SERVO_BINARY_PATH=target/${{ inputs.profile }}/servo/servoshell-default-signed.hap" >> $GITHUB_ENV
+      - name: File location (Open Harmony, unsigned)
+        continue-on-error: true
+        if: ${{ contains(inputs.target, 'ohos') && env.SERVO_OHOS_SIGNING_CONFIG == '' }}
+        run: echo "SERVO_BINARY_PATH=target/${{ inputs.profile }}/servo/servoshell-default-unsigned.hap" >> $GITHUB_ENV
```

Comment:
Not quite happy about this part with tons of platform hardcoded setup, while we can definitely just pass binary path as input, still doesn't quite solve the complication of passing whether the Open Harmony artifact is signed or not.